### PR TITLE
fix: If country on cart has no tax region, clear tax lines

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**

If the country on the shipping address changes to a country without an associated tax region, we clear the tax lines on shipping methods and line items.

### CODE REVIEW

File: packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts [hunk 1]
```diff
@@ -35,18 +35,15 @@ export const setTaxLinesForItemsStep = createStep(
...
     const shippingTaxLinesData =
       normalizeShippingTaxLinesForCart(shipping_tax_lines)
-    const setShippingTaxLinesPromise = shippingTaxLinesData.length
-      ? await cartService.setShippingMethodTaxLines(
-          cart.id,
-          shippingTaxLinesData
-        )
-      : 0
+    const setShippingTaxLinesPromise =
+      await cartService.setShippingMethodTaxLines(cart.id, shippingTaxLinesData)
```
Comment (olivermrbl) [inline packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts]: In the case of no tax region for the provided country, `shippingTaxLinesData` and `itemsTaxLinesData` will be empty arrays and that should be reflected on the cart.

**Review by fPolic — APPROVED** [review]

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts (hunk 1 above)
Comment (riqwan) [inline packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts]: No longer a promise, can be removed from the promise array.

```suggestion
    await cartService.setShippingMethodTaxLines(cart.id, shippingTaxLinesData)
```

**Review by riqwan — APPROVED** [review]
lgtm!

File: packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts]: I guess none of the promises are relevant. Please check latest commit.

**Review by riqwan — APPROVED** [review]

**Pull request merged**